opensquirrel/ir.py

```diff
@@ -140,23 +140,38 @@ def accept(self, visitor: IRVisitor) -> Any:
         return visitor.visit_bit(self)
 
 
-@dataclass
+@dataclass(init=False)
 class Qubit(Expression):
+    """``Qubit`` is used for intermediate representation of ``Statement`` arguments.
+
+    Attributes:
+        index: index of the ``Qubit`` object.
+    """
+
     index: int
 
+    def __init__(self, index: QubitLike) -> None:
```

Comment:
Maybe `value` a better name than `index` here.

Reply:
I disagree, if I think of a qubit value, I immediately think of the ket state.

Reply:
Fair enough. I thought of `qubit`, but the `__init__` parameter is actually an index value. I only wanted to avoid the `index.index`, which sounded a bit weird to me.